fontforgeexe/scriptingdlg.c

```diff
@@ -190,9 +190,7 @@ return( true );
     if ( event->type==et_close ) {
 	SD_DoCancel( sd );
     } else if ( event->type==et_charup ) {
-    char* contents = GGadgetGetTitle8(GWidgetGetControl(sd->gw,CID_Script));
-    sd->fv->script_unsaved = (bool)strlen(contents);
-    free(contents);
+    sd->fv->script_unsaved = !GTextFieldIsEmpty(GWidgetGetControl(sd->gw,CID_Script));
     } else if ( event->type==et_save ) {
```

Comment:
surely this would need changing as its now a subtype of `et_controlevent`. Also what happened to listening for the text changed event? There are other ways it can change (e.g. right click paste)